staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/authorize.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2020 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"errors"
+
+	authentication "k8s.io/api/authentication/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core"
+)
+
+const (
+	// AnnPrePopulated is a PVC annotation telling the datavolume controller that the PVC is already populated
+	AnnPrePopulated = core.GroupName + "/storage.prePopulated"
+	// AnnCheckStaticVolume checks if a statically allocated PV exists before creating the target PVC.
+	// If so, PVC is still created but population is skipped
+	AnnCheckStaticVolume = core.GroupName + "/storage.checkStaticVolume"
+)
+
+// ErrNoTokenOkay indicates proceeding without token is allowed
+// This error should only be of interest to entities that give out DataVolume tokens
+var ErrNoTokenOkay = errors.New("proceeding without token is okay under the circumstances")
+
+// AuthorizeUser indicates if the creating user is authorized to create the data volume
+// For sources other than clone (import/upload/etc), this is a no-op
+func (dv *DataVolume) AuthorizeUser(requestNamespace, requestName string, proxy AuthorizationHelperProxy, userInfo authentication.UserInfo) (bool, string, CloneSourceHandler, error) {
```

Comment:
Rather than returning four vales, how about a new struct that combines the first three?

Reply:
Yep makes sense, added
My biggest worry is still this though:
https://github.com/kubevirt/containerized-data-importer/pull/2636#discussion_r1165542038

Reply:
I think it's fine to delete auth.go.  I seriously doubt any project other than kubevirt is importing it

Reply:
awesome